test/test_utility_functions.py

```diff
@@ -151,3 +152,17 @@ def test_does_not_increment_without_stream_send(self):
 
         third_stream_id = c.get_next_available_stream_id()
         assert third_stream_id == (first_stream_id + 2)
+
+
+class TestExtractHeader(object):
+
+    example_request_headers = [
+        (':authority', 'example.com'),
+        (':path', '/'),
+        (':scheme', 'https'),
+        (':method', 'GET'),
+    ]
+
+    def test_extract_header_method(self):
+        assert extract_method_header(self.example_request_headers) \
+            == u'GET'
```

Comment:
In addition to this, I think I'd like two tests in the main tests to confirm that we use this information correctly. Firstly, one test showing a HEAD request and response with a non-zero content length and no body that works just fine. Then a test showing a HEAD request and a response with a non-zero content length and a body that is correctly rejected.

Do you feel up to writing those two tests? You can take inspiration from some of the other tests in`test_basic_logic`.